### PR TITLE
Added a first class field for serviceName

### PR DESCRIPTION
**Pull request by anujgoyal1**

### CODE REVIEW

File: data-model/src/main/avro/Event.avdl [hunk 1]
```diff
@@ -56,5 +56,7 @@ protocol EventProtocol {
     union { null, org.hypertrace.core.datamodel.eventfields.http.Http } http = null;
     union { null, org.hypertrace.core.datamodel.eventfields.grpc.Grpc } grpc = null;
     union { null, org.hypertrace.core.datamodel.eventfields.sql.Sql } sql = null;
+
+    union { null, string } service_name = null;
```
Comment (tim-mwangi) [inline data-model/src/main/avro/Event.avdl:60]: Have we decided to go this route?

Comment (kotharironak) [reply data-model/src/main/avro/Event.avdl:60]: @tim-mwangi This is what this ticket - https://traceableai.atlassian.net/browse/ENG-4585 mentioned. But, let a @buchi-busireddy comment.

Comment (buchi-busireddy) [reply data-model/src/main/avro/Event.avdl:60]: Yes.

Comment (skjindal93) [reply data-model/src/main/avro/Event.avdl:60]: We already have service name populated inside `jaegerFields` on the Event. Do we also need it outside?

Comment (buchi-busireddy) [reply data-model/src/main/avro/Event.avdl:60]: Correct. @anujgoyal1 this field will replace the `service_name` from `jaegerFields` so we should deprecate that in avro compatible way while adding the new field.

File: data-model/src/main/avro/eventfields/jaeger/JaegerFields.avdl [hunk 2]
```diff
@@ -4,6 +4,5 @@ protocol JaegerFieldsProtocol {
     int flags = 0;
     union { null, array<string> } logs = null;
     union { null, array<string> } warnings = null;
-    union { null, string } service_name = null;
```
Comment (anujgoyal1) [inline data-model/src/main/avro/eventfields/jaeger/JaegerFields.avdl:7]: Removing the ```service_name``` . We looked into usages, everywhere it is being consumed using the constant 
```JaegerAttribute.JAEGER_ATTRIBUTE_SERVICE_NAME``` . Setter is being used in ```JaegerSpanNormaliser``` in repo ```span-normaliser``` which will be removed.

I'm assuming it's a action. 
cc: @buchi-busireddy @kotharironak

**Pull request merged**